src/core/task/task.h

```diff
@@ -96,9 +98,19 @@ class LegateTask {
     Core::show_progress(task, legion_context, runtime, task_name());
 
     TaskContext context(task, regions, legion_context, runtime);
-    if (!Core::use_empty_task) (*TASK_PTR)(context);
 
-    return context.pack_return_values();
+    try {
+      if (!Core::use_empty_task) (*TASK_PTR)(context);
+      return context.pack_return_values();
+    } catch (legate::TaskException& e) {
+      if (context.can_raise_exception()) {
+        context.make_all_unbound_stores_empty();
+        return context.pack_return_values_with_exception(e.index(), e.error_message());
+      } else
+        // If a Legate exception is thrown by a task that does not declare any exception,
+        // this is a bug in the library that needs to be reported to the developer
+        Core::report_unexpected_exception(task_name(), e);
+    }
```

Comment:
clang warns here that
```
/Users/mpapadakis/legate.core/install/include/core/task/task.h:114:3: warning: non-void function does not return a value in all control paths [-Wreturn-type]
```

Reply:
a4463afbdcac4549d3d83618fadfea5f9f7535e7